homeassistant/loader.py

```diff
@@ -54,7 +59,29 @@
 MAX_LOAD_CONCURRENTLY = 4
 
 
-def manifest_from_legacy_module(domain: str, module: ModuleType) -> Dict:
+class Manifest(TypedDict, total=False):
+    """Integration manifest."""
+
+    name: str
+    disabled: Optional[str]
```

Comment:
If `disabled` is present in the dict, will it always be a string?

Reply:
I think so, it was already indirectly hinted as such before these changes:
```python
     @property
     def disabled(self) -> Optional[str]:
         """Return reason integration is disabled."""
         return cast(Optional[str], self.manifest.get("disabled"))
```

Reply:
So then we shouldn't type it as `Optional`, right? We can add a comment that it's optional or maybe comment the other ones that are required.

Reply:
Ah, you meant that. Guess we could go into looking for and making such changes here too, however my intention was to not make any changes of that nature, but just clean up and improve on missing parts. (Kind of an exception to that are `zeroconf` and `dhcp` which had incorrect `cast`s, but the casts went away entirely when moving to `TypedDict`.)

Thoughts?

Reply:
We're not changing existing code. We're correctly typing the data in the TypedDict. The property `disabled` returns `Optional[str]` since the lookup of the manifest may return None if the key is missing. The `disabled`item in the manifest data though should not be typed as `Optional` since it should never be `None` if present.

Reply:
It can be argued both ways, and neither is 100% correct. TypedDict cannot specify which specific attributes are required present, which not, and to preserve the "specification" which of them we want present, I chose to mark the non-required ones as Optional. But I don't feel that strongly about it so changed that field, ditto all others, as non-Optional, as this is not just about `disabled`.

Doing so required removing the legacy `"documentation": None`, grepping through the tree I found nothing that would break as a result.